### ISSUE TITLE
Identical Devices (ZY-M201-WIFI24G) one seen as mmwave and one seen as PIR

### ISSUE THREAD

**Issue by haroldboom** [make-all/tuya-local#1515]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**
Identical Devices (ZY-M201-WIFI24G) one seen as correctly mmwave and one seen incorrectly pir_sensor or a curtain, both have identical firmware versions. Added about a week apart,

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:

Add a couple of ZY-M201-WIFI24G and see what they are recognized as

**Expected behavior**

Both work identically I would think.

**Additional context**

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

Device matches pir_sensor with quality of 33%. DPS: {"updated_at": 1705033391.6767554, "1": "none", "4": 550, "9": 0, "101": "nobody", "102": 5, "104": 24}

**Comment by haroldboom** [comment]
A factory reset of the device and re-add to Tuya cloud fixed the issue

**Comment by make-all** [comment]
Although you have solved your immediate problem, the underlying issue of the sensitivity settings sometimes being missing from the device's reported data still worth accounting for in the config.

**Comment by vivaleisure** [comment]
same issue here, have 3 of these ZY-M201's all detected as mmwave, and now the next 3 i need to install are all detecting as either  pir_sensor or loratap_curtain_switch

my scan is coming up as:
   [WIFI24G Presence Sensor 5] 192.168.155.98     - [Off] - DPS: {'1': 'move', '4': 550, '9': 150, '101': 'exist', '102': 5, '104': 44}

tried the factory reset and re-adding to cloud and app, but no good.

**Comment by haroldboom** [comment]
> same issue here, have 3 of these ZY-M201's all detected as mmwave, and now the next 3 i need to install are all detecting as either pir_sensor or loratap_curtain_switch
> 
> my scan is coming up as: [WIFI24G Presence Sensor 5] 192.168.155.98 - [Off] - DPS: {'1': 'move', '4': 550, '9': 150, '101': 'exist', '102': 5, '104': 44}
> 
> tried the factory reset and re-adding to cloud and app, but no good.

Hey mate, try and do the “Factory reset and remove from account” option in the app but also do a hard reset on the device. Hold down the reset button underneath the sensor until it flashes

**Comment by vivaleisure** [comment]
thanks for that.. i tried the factory reset and it didn't work

what i I just did that fixed it was by 'setting every setting' via the... so i just powered it down and up (no factory reset required), and just set the sensitivities, and all other settings.  it seems when it boots with defaults it doesn't set all these...

confirmed this with one of the other devices that wasn't working earlier.

this is what a scan showed before i set the settings - ie: added to cloud and app, and tried to add to HA
    Status: {'1': 'move', '4': 250, '9': 150, '101': 'exist', '102': 5, '104': 8}

and as soon as i went and set all the settings to random values in the app
    Status: {'1': 'none', '2': 4, '4': 250, '9': 0, '101': 'nobody', '102': 60, '104': 12, '111': 6}

so a few more 'sensors' came up.

**Comment by haroldboom** [comment]
> thanks for that.. i tried the factory reset and it didn't work
> 
> what i I just did that fixed it was by 'setting every setting' via the... so i just powered it down and up (no factory reset required), and just set the sensitivities, and all other settings. it seems when it boots with defaults it doesn't set all these...
> 
> confirmed this with one of the other devices that wasn't working earlier.
> 
> this is what a scan showed before i set the settings - ie: added to cloud and app, and tried to add to HA Status: {'1': 'move', '4': 250, '9': 150, '101': 'exist', '102': 5, '104': 8}
> 
> and as soon as i went and set all the settings to random values in the app Status: {'1': 'none', '2': 4, '4': 250, '9': 0, '101': 'nobody', '102': 60, '104': 12, '111': 6}
> 
> so a few more 'sensors' came up.

Ah you know what this is probably what I accidentally did that fixed it as well, I was messing with all the settings in the app

**Comment by mad-tunes** [comment]
I just found I was only given incorrect device types as I tried to add this device to a 2nd/replacement HA install, no 'zym100 presence sensor'.
Can confirm that factory reset through the Tuya IOT site or removing it from the smartlife app and re-adding it didn't change anything.
Re-applying the sensors settings through smart life did the trick though, then when I added it it was shown as zym100_presence_sensor again and works a treat.
Thanks for sharing the trick
